### PR TITLE
[VL-93] Change check notification status endpoint

### PR DESCRIPTION
**Pull request by datalek**
Before this commit, the check notification status endpoint always returned the notification in the WAITING state. After this commit, the endpoint returns the notification with a different state. The new returned status is based on how many times the endpoint was called.

#### How Has This Been Tested?

Unit tests and e2e tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by datalek** [comment]
This PR has a connection to #37. We should update one of the two to incorporate the changes once one of them is merged

**Comment by gunzip** [comment]
We use [dpulls](https://www.dpulls.com/) so you can write "Depends on #37" I guess.

**Comment by datalek** [comment]
> We use [dpulls](https://www.dpulls.com/) so you can write "Depends on #37" I guess.

Nice! Depends on https://github.com/pagopa/pn-local-emulator/pull/37

File: src/domain/CheckNotificationStatusRepository.ts [hunk 1]
```diff
@@ -15,20 +17,37 @@ export type CheckNotificationStatusRecord = {
...
-      return O.some({
-        ...record.output.returned,
-        ...record.input.body,
-        notificationRequestStatus: '__WAITING__',
-      });
-    default:
-      return O.none;
-  }
-};
+export const getNewNotificationRequestStatusResponse = (
```
Comment (kin0992) [inline src/domain/CheckNotificationStatusRepository.ts:20]:
```suggestion
const getNewNotificationRequestStatusResponse = (
```
this method is only called on line 25

File: src/domain/NewNotificationRepository.ts [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 
+export type Notification = NewNotificationRequest & NewNotificationResponse;
+
 export type NewNotificationRecord = {
   type: 'NewNotificationRecord';
   input: { apiKey: ApiKey; body: NewNotificationRequest };
   output: Response<202, NewNotificationResponse> | Response<403, UnauthorizedMessageBody>;
 };
 
+export const getNotification = (record: NewNotificationRecord): O.Option<Notification> =>
```
Comment (kin0992) [inline src/domain/NewNotificationRepository.ts:17]:
```suggestion
const getNotification = (record: NewNotificationRecord): O.Option<Notification> =>
```
private method

**Review by kin0992 — CHANGES_REQUESTED** [review]

**Review by kin0992 — APPROVED** [review]
🌳

**Pull request merged**